### ISSUE TITLE
Is is safe to embed JSON in CUE?

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#20]
_Originally opened by @enisoc in https://github.com/cuelang/cue/issues/20_

Since CUE is a superset of JSON, I understand that you can paste valid JSON into a `.cue` file and it will be guaranteed valid CUE as well.

My question is, is it guaranteed that valid JSON pasted into a `.cue` file will *never* accidentally invoke/trigger any features of CUE that are not in JSON?

For example, I'm guessing that string interpolation can't be triggered accidentally by valid JSON, since (I presume) the escape sequence `\(` is invalid JSON. Is CUE designed in such a way that all such features are systematically impossible to trigger from valid JSON?

In other words, is all valid JSON guaranteed to produce exactly the same output when treated as a CUE program? I haven't seen mention of this in the docs.

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/20#issuecomment-464831931_

Yes, valid JSON should produce exactly the same result. 

`\(` is not valid JSON

However, there are a few small discrepancies that need fixing or validation. There is a CL pending to support `\/`. 
Surrogate pairs are currently disallowed in strings. I should allow them, but it is a bit unclear how to interpret both valid and invalid pairs within the language. Probably I should translate valid pairs to their corresponding code points at least.

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/20#issuecomment-480781561_

Escaped solidus and surrogate pairs are now supported. Non-escaped surrogate values are not allowed by the Go decoder and are currently still not supported. This is technically correct according to the Unicode standard and is allowed by the RFC 4627 (JSON), but this may break some setups in practice for non-conforming input. This can be changed if need be.

Other than that, all valid JSON should now be interpreted exactly as defined.
